narwhal/consensus/src/utils.rs

```diff
@@ -19,7 +19,7 @@ where
 {
     let mut to_commit = vec![leader.clone()];
     let mut leader = leader;
-    for r in (state.last_committed_round + 2..leader.round())
+    for r in (state.last_committed_round + 2..=leader.round() - 2)
```

Comment:
Why this change? Isn't it the same thing?

Reply:
@asonnino should I close this, or is this a no-op?

Reply:
oh I missed your comment, checking now

Reply:
That seems a bug (the `rev` got me confused), a dangerous one

Reply:
I see. Will try to add more tests to catch issues like this.